src/main/java/org/opensearch/flowframework/model/WorkflowState.java

```diff
@@ -314,6 +336,9 @@ public static WorkflowState updateExistingWorkflowState(WorkflowState existingSt
         if (stateWithNewFields.resourcesCreated() != null) {
             builder.resourcesCreated(stateWithNewFields.resourcesCreated());
         }
+        if (stateWithNewFields.getTenantId() != null) {
```

Comment:
should this cover bwc issues? Should we test for that manually or with a test as a TODO at least?

Reply:
I don't understand the question here.  without tenant awareness this will probably always be null anyway.  Our unit tests use null nearly everywhere.  Tenant aware integ tests test this thoroughly when it's set.

Reply:
I see you covered this bwc in other comments